CRM/Core/BAO/FinancialTrxn.php

```diff
@@ -696,4 +696,42 @@ public static function createDeferredTrxn($lineItems, $contributionDetails, $upd
     }
   }
 
+  /**
+   * Update Credit Card Details in civicrm_financial_trxn table.
+   *
+   * @param array $params
+   *   Contribution params
+   *
+   */
+  public static function updateCreditCardDetails($params) {
+    $financialTrxn = civicrm_api3('EntityFinancialTrxn', 'get', array(
+      'return' => array('financial_trxn_id.payment_processor_id', 'financial_trxn_id'),
+      'entity_table' => 'civicrm_contribution',
+      'entity_id' => $params['contribution']->id,
+      'financial_trxn_id.is_payment' => TRUE,
+      'options' => array('sort' => 'financial_trxn_id DESC', 'limit' => 1),
+    ));
+
+    if (!$financialTrxn['count']) {
```

Comment:
In case of Contribution status is Pending From Incomplete Transaction or Failed there is no Financial Entries created for Contribution.